### PR TITLE
New/14.0 workflow close project automatically when its tasks are all done

### PR DESCRIPTION
**Pull request by atm-florianm**
# NEW
New workflow option: whenever the progress of all tasks in a project reach 100%, close the project automatically. Note: as it stands, even a draft project can be closed that way; this option is non-retroactive (when enabled, it won't loop over all existing projects and close those meeting the criterion: it will only check a project when one of its tasks is modified).

### CODE REVIEW

File: htdocs/core/triggers/interface_20_modWorkflow_WorkflowManager.class.php [hunk 1]
```diff
@@ -315,6 +316,39 @@ public function runTrigger($action, $object, User $user, Translate $langs, Conf
 			}
 		}
 
+		// classify project as closed when progress of all tasks is 100%
+		if ($action == 'TASK_MODIFY') {
+			//
+			if (!empty($conf->projet->enabled) && !empty($conf->global->WORKFLOW_PROJECT_CLASSIFY_CLOSED_WHEN_ALL_TASKS_DONE))
```
Comment (atm-florianm) [inline htdocs/core/triggers/interface_20_modWorkflow_WorkflowManager.class.php:322]: Should I also test whether the project is validated?

File: htdocs/admin/workflow.php [hunk 2]
```diff
@@ -137,7 +148,15 @@
 		'position' => 66,
 		'enabled' => ! empty($conf->expedition->enabled) && ! empty($conf->facture->enabled),
 		'picto' => 'shipment'
-	)
+	),
+
+	// Automatic classification of projects
+	'WORKFLOW_PROJECT_CLASSIFY_CLOSED_WHEN_ALL_TASKS_DONE' => array(
```
Comment (eldy) [inline htdocs/admin/workflow.php:154]: The "workflow" module is dedicated to manage interaction between different modules.
But here project and class are in the same module. So please move the option into the setup of module project.

File: htdocs/core/triggers/interface_20_modWorkflow_WorkflowManager.class.php (hunk 1 above)
Comment (eldy) [reply htdocs/core/triggers/interface_20_modWorkflow_WorkflowManager.class.php:322]: I think so. A draft must stay a draft.

**Comment by atm-florianm** [comment]
Closed and made anew, cf. https://github.com/Dolibarr/dolibarr/pull/16312

**Pull request closed without merging**